homeassistant/components/wallbox/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""The Wallbox integration."""
+import asyncio
+import logging
+
+import requests
+from wallbox import Wallbox
+
+from homeassistant import exceptions
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_CONNECTIONS, CONF_STATION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+class WallboxHub:
+    """Wallbox Hub class."""
+
+    def __init__(self, station, username, password):
+        """Initialize."""
+        self._station = station
+        self._username = username
+        self._password = password
+
+    async def async_authenticate(self, hass) -> bool:
+        """Authenticate using Wallbox API."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
```

Comment:
OK, what do you mean exactly, because this already is a method in WallboxHub? Or do you mean to create a 'private' method which is not async but does the try catch part and then a public method to call that? Or do you mean something entirely different?

Reply:
I mean keep async_authenticate in WallboxHub as the public interface. The try/except should not contain async_add_executor_job

new method 
def _authenticate():
       try:
            wallbox = WallBox …. the library
            wallbox.authenticate()
       except

and in async_authenticate
         await hass.async_add_executor_job(self._authenticate)

that try/except and library call run in the same thread.

Reply:
yes, I understand now. Done